assets/src/edit-story/components/inspector/prepublish/autoVideoOptimization.js

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useEffect, useRef } from 'react';
+import { __, TranslateWithMarkup } from '@web-stories-wp/i18n';
+import { trackClick } from '@web-stories-wp/tracking';
+
+/**
+ * Internal dependencies
+ */
+import { Link, THEME_CONSTANTS } from '../../../../design-system';
+import { useConfig } from '../../../app';
+import {
+  DescriptionText,
+  PageIndicator,
+  StyledToggle,
+  ToggleGroup,
+  VideoOptimizationGroup,
+} from './styles';
+
+const AutoVideoOptimization = ({
+  areVideosAutoOptimized,
+  onAutoOptimizeVideoClick,
+}) => {
+  const { dashboardSettingsLink } = useConfig();
+
+  const wereVideosInitiallyAutoOptimized = useRef(areVideosAutoOptimized);
+
+  useEffect(() => {
+    // areVideosAutoOptimized may be undefined initially since we
+    // are waiting on an api call. Once that is done, we can see the
+    // value of the `areVideosAutoOptimized`
+    if (
+      areVideosAutoOptimized &&
+      wereVideosInitiallyAutoOptimized.current === undefined
+    ) {
+      wereVideosInitiallyAutoOptimized.current = true;
+    }
+  }, [areVideosAutoOptimized]);
+
+  if (wereVideosInitiallyAutoOptimized.current) {
+    return null;
+  }
+
+  return (
+    <VideoOptimizationGroup>
+      <PageIndicator>{__('General', 'web-stories')}</PageIndicator>
```

Comment:
This is great for right now because I don't want to add more to our scope than we already have 😅  but something to think about - the next version of this panel will have a `general` section regardless of if the optimize toggle is present or not, kind of like a `page` subsection of the checklist, so we'll need to probably make it mappable with its contents. This is absolutely not part of this work here, i just wanted to mention it.

Reply:
Yeah - this will definitely have to change!